### ISSUE TITLE
[oauthproxy.go:871] Error loading cookied session

### ISSUE THREAD

**Issue by jemega** [oauth2-proxy/oauth2-proxy#717]
I have implemented oauth2 proxy provider keycloak + Nginx ingress for an application hosted in AKS.

All the services in k8s are running well if I trying to reach the services through ingress works, but if I try to reach through an Application gateway + oauth2 proxy in the browser don't redirect to the correct URL and return a 500 Error

## Expected Behavior

Reach all the services in k8s throw Application gateway + oauth2 proxy (provider keycloak)

## Current Behavior

Status code: 500

Error of oauth2 proxy service: 
```
[oauthproxy.go:871] Error loading cookied session: Cookie "_oauth2_proxy" not present
...
[oauthproxy.go:871] Error loading cookied session: Cookie Signature not valid
```

oauth proxy config:
```
upstream= [
  "https://myhost"
]

provider="keycloak"

login_url="https://myhost/auth/realms/myrealm_keycloak/protocol/openid-connect/auth"
redeem_url="https://myhost/auth/realms/myrealm_keycloak/protocol/openid-connect/token"
validate_url="https://myhost/auth/realms/myrealm_keycloak/protocol/openid-connect/userinfo"
keycloak_group=mygroup_keycloak
email_domains=[ "*" ]

client_id= client_keycloak
client_secret= clientsecret_keycloak
# cookie secret generated using the command 
# python -c 'import os,base64; print base64.b64encode(os.urandom(16))'
cookie_secret= cookiesecret
```

## Your Environment

The oauth2 proxy is installed in VM separated of AKS cluster in azure. Keycloak has a Valid Redirect URIs https://myhost/oauth2/callback

- Version used: v5.1.1

**Comment by NickMeves** [comment]
Not positive if this is the issue, but 5.1.1 had a cookie secret padding bug that was fixed with this PR in v6.0.0: https://github.com/oauth2-proxy/oauth2-proxy/pull/556

Try doing `python -c 'import os,base64; print base64.b64encode(os.urandom(24))'`

24 bytes will base64 encode to 32 bytes.  So even if the base64 decoding padding bug rears its head, you end up with either 24 or 32 bytes, both of which are valid for the AES cipher usage (must be 16, 24, or 32 byte length for AES 128/192/256 bit)

**Comment by jemega** [comment]
About the issue, I don't sure what happened, but with the following creation of cookie secrets works

`docker run -ti --rm python:3-alpine python -c 'import secrets,base64; print(base64.b64encode(base64.b64encode(secrets.token_bytes(16))));'`
